### PR TITLE
awscli: 1.19.52 -> 1.19.55

### PR DESCRIPTION
**Pull request by NeQuissimus**
###### Motivation for this change
Updates

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120066` at ace3ea60 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>awsebcli</li>
    <li>mlflow-server</li>
    <li>python38Packages.flower</li>
    <li>python38Packages.geopandas</li>
    <li>python38Packages.osmnx</li>
    <li>python39Packages.flower</li>
    <li>python39Packages.geopandas</li>
    <li>python39Packages.osmnx</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/7b056e7588b44120fd00e01886de282c">awscli2</a></li>
    <li><a href="https://gist.github.com/bb78ff406dc9587b6d0a17e647e8fd81">python38Packages.cartopy</a></li>
    <li><a href="https://gist.github.com/da38449e4e2ebaee48ecce8a628af771">python38Packages.fiona</a></li>
    <li><a href="https://gist.github.com/80306b892a5eae9a9b8b440ff75d6fc3">python39Packages.cartopy</a></li>
    <li><a href="https://gist.github.com/8ea2ec6704ed65681897c534612fa93d">python39Packages.fiona</a></li>
  </ul>
</details>
<details>
  <summary>4 packages skipped due to time constraints:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.tensorboardx</li>
    <li>python39Packages.tensorboardx</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>115 packages built successfully:</summary>
  <ul>
    <li>aws-adfs (python38Packages.aws-adfs)</li>
    <li>aws-google-auth</li>
    <li>aws-mfa</li>
    <li>aws-sam-cli</li>
    <li>awscli</li>
    <li>awslogs</li>
    <li>barman</li>
    <li>checkov</li>
    <li>cloud-custodian</li>
    <li>credstash (python38Packages.credstash)</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>dvc-with-remotes</li>
    <li>elasticsearch-curator</li>
    <li>git-remote-codecommit</li>
    <li>iceshelf</li>
    <li>lexicon</li>
    <li>luigi</li>
    <li>mapproxy</li>
    <li>nixops</li>
    <li>python38Packages.aiobotocore</li>
    <li>python38Packages.aws-sam-translator</li>
    <li>python38Packages.aws-xray-sdk</li>
    <li>python38Packages.boto3</li>
    <li>python38Packages.botocore</li>
    <li>python38Packages.celery</li>
    <li>python38Packages.certbot-dns-route53</li>
    <li>python38Packages.cfn-lint</li>
    <li>python38Packages.chalice</li>
    <li>python38Packages.django-anymail</li>
    <li>python38Packages.django-raster</li>
    <li>python38Packages.djmail</li>
    <li>python38Packages.ec2instanceconnectcli</li>
    <li>python38Packages.flowlogs_reader</li>
    <li>python38Packages.fs-s3fs</li>
    <li>python38Packages.gensim</li>
    <li>python38Packages.gradient</li>
    <li>python38Packages.hass-nabucasa</li>
    <li>python38Packages.kombu</li>
    <li>python38Packages.labelbox</li>
    <li>python38Packages.mapbox</li>
    <li>python38Packages.moto</li>
    <li>python38Packages.pycognito</li>
    <li>python38Packages.pynamodb</li>
    <li>python38Packages.rasterio</li>
    <li>python38Packages.s3fs</li>
    <li>python38Packages.s3transfer</li>
    <li>python38Packages.sagemaker</li>
    <li>python38Packages.sapi-python-client</li>
    <li>python38Packages.scrapy</li>
    <li>python38Packages.scrapy-deltafetch</li>
    <li>python38Packages.scrapy-fake-useragent</li>
    <li>python38Packages.scrapy-splash</li>
    <li>python38Packages.serverlessrepo</li>
    <li>python38Packages.slack-sdk</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.smart_open</li>
    <li>python38Packages.snowflake-connector-python</li>
    <li>python38Packages.snowflake-sqlalchemy</li>
    <li>python38Packages.warrant</li>
    <li>python39Packages.aiobotocore</li>
    <li>python39Packages.aws-adfs</li>
    <li>python39Packages.aws-sam-translator</li>
    <li>python39Packages.aws-xray-sdk</li>
    <li>python39Packages.boto3</li>
    <li>python39Packages.botocore</li>
    <li>python39Packages.celery</li>
    <li>python39Packages.certbot-dns-route53</li>
    <li>python39Packages.cfn-lint</li>
    <li>python39Packages.chalice</li>
    <li>python39Packages.credstash</li>
    <li>python39Packages.django-anymail</li>
    <li>python39Packages.django-raster</li>
    <li>python39Packages.djmail</li>
    <li>python39Packages.ec2instanceconnectcli</li>
    <li>python39Packages.flowlogs_reader</li>
    <li>python39Packages.fs-s3fs</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.hass-nabucasa</li>
    <li>python39Packages.kombu</li>
    <li>python39Packages.labelbox</li>
    <li>python39Packages.mapbox</li>
    <li>python39Packages.moto</li>
    <li>python39Packages.pycognito</li>
    <li>python39Packages.pynamodb</li>
    <li>python39Packages.rasterio</li>
    <li>python39Packages.s3fs</li>
    <li>python39Packages.s3transfer</li>
    <li>python39Packages.sagemaker</li>
    <li>python39Packages.sapi-python-client</li>
    <li>python39Packages.scrapy</li>
    <li>python39Packages.scrapy-deltafetch</li>
    <li>python39Packages.scrapy-fake-useragent</li>
    <li>python39Packages.scrapy-splash</li>
    <li>python39Packages.serverlessrepo</li>
    <li>python39Packages.slack-sdk</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.snowflake-connector-python</li>
    <li>python39Packages.snowflake-sqlalchemy</li>
    <li>python39Packages.warrant</li>
    <li>s3bro</li>
    <li>s4cmd</li>
    <li>snakemake</li>
    <li>sourcehut.buildsrht</li>
    <li>sourcehut.dispatchsrht</li>
    <li>sourcehut.gitsrht</li>
    <li>sourcehut.hgsrht</li>
    <li>sourcehut.hubsrht</li>
    <li>sourcehut.listssrht</li>
    <li>sourcehut.mansrht</li>
    <li>sourcehut.metasrht</li>
    <li>sourcehut.pastesrht</li>
    <li>sourcehut.todosrht</li>
    <li>streamlit</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/s3transfer/default.nix:47:3:
```
   |
47 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/s3transfer/default.nix:50:0:
```
   |
50 |     description = "A library for managing Amazon S3 transfers";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/boto3/default.nix:40:3:
```
   |
40 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/botocore/default.nix:43:3:
```
   |
43 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by SuperSandro2000** [comment]
awscli2 fails with 

```
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./awscli-2.1.35-py2.py3-none-any.whl
Requirement already satisfied: distro<1.6.0,>=1.5.0 in /nix/store/kd59ghxmifq8fj0k80hmbns4f41fzraz-python3.8-distro-1.5.0/lib/python3.8/site-packages (from awscli==2.1.35) (1.5.0)
Requirement already satisfied: ruamel.yaml>=0.15.0 in /nix/store/4p6z5v3s8wgv41b4x60b6q6pgimw44c6-python3.8-ruamel.yaml-0.16.12/lib/python3.8/site-packages (from awscli==2.1.35) (0.16.12)
Requirement already satisfied: prompt-toolkit<3.0.0,>=2.0.0 in /nix/store/cfgb2zj1hpf21hfiad1jjvmykp255qdw-python3.8-prompt_toolkit-2.0.10/lib/python3.8/site-packages (from awscli==2.1.35) (2.0.10)
Requirement already satisfied: colorama>=0.2.5 in /nix/store/hmyyhfcrangfi8y990sfrj5pdlb15lx0-python3.8-colorama-0.4.4/lib/python3.8/site-packages (from awscli==2.1.35) (0.4.4)
Requirement already satisfied: cryptography>=3.3.2 in /nix/store/xvjl332g2ifjrzmwkl83yrxhqhjgz0ls-python3.8-cryptography-3.4.7/lib/python3.8/site-packages (from awscli==2.1.35) (3.4.7)
Requirement already satisfied: docutils>=0.10 in /nix/store/b2aq4x6ynqj0fbsx9lnh0wbc8a5hwsnc-python3.8-docutils-0.16/lib/python3.8/site-packages (from awscli==2.1.35) (0.16)
Requirement already satisfied: botocore==2.0.0dev103 in /nix/store/7hsi9bscdp71p99sbx8s2h8bzfjd8yvg-python3.8-botocore-2.0.0dev103/lib/python3.8/site-packages (from awscli==2.1.35) (2.0.0.dev103)
ERROR: Could not find a version that satisfies the requirement s3transfer<0.4.0,>=0.3.0 (from awscli)
ERROR: No matching distribution found for s3transfer<0.4.0,>=0.3.0
```

File: pkgs/development/python-modules/s3transfer/default.nix [hunk 1]
```diff
@@ -14,15 +14,16 @@
...
-  version = "0.3.4";
+  version = "0.4.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "7fdddb4f22275cf1d32129e21f056337fd2a80b6ccef1664528145b72c49e6d2";
+    sha256 = "sha256-89/XkcrSeZQD48gFGBCnym7h0uYw5dKo+WSdiSvbPbY=";
   };
 
   propagatedBuildInputs =
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/s3transfer/default.nix:24]:
```suggestion
  propagatedBuildInputs = [
```

File: pkgs/development/python-modules/s3transfer/default.nix [hunk 2]
```diff
@@ -14,15 +14,16 @@
...
-    [ botocore
+    [
+      botocore
     ] ++ lib.optional (pythonOlder "3") futures;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/s3transfer/default.nix:27]:
```suggestion
    botocore
  ] ++ lib.optionals (pythonOlder "3") [ futures ];
```

**Comment by NeQuissimus** [comment]
awscli2 needs an update to 2.2.0, I will get around to that sometime this week. 2.2.0 has a new dependency, which we don't have in nixpkgs.

File: pkgs/development/python-modules/botocore/default.nix [hunk 3]
```diff
@@ -13,11 +13,11 @@
 
 buildPythonPackage rec {
   pname = "botocore";
-  version = "1.20.52"; # N.B: if you change this, change boto3 and awscli to a matching version
+  version = "1.20.57"; # N.B: if you change this, change boto3 and awscli to a matching version
```
Comment (jonringer) [inline pkgs/development/python-modules/botocore/default.nix:16]: looks like awscli doesn't like this
```
builder for '/nix/store/sza90gqk2h0c6s0c6zjbnsf9zfb960c7-awscli-1.19.52.drv' failed with exit code 1; last 10 log lines:
  adding 'awscli-1.19.52.dist-info/top_level.txt'
  adding 'awscli-1.19.52.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/awscli-1.19.52/dist /build/awscli-1.19.52
  Processing ./awscli-1.19.52-py2.py3-none-any.whl
  ERROR: Could not find a version that satisfies the requirement botocore==1.20.52 (from awscli)
  ERROR: No matching distribution found for botocore==1.20.52
  ```

**Review by jonringer — COMMENTED** [review]
o

EDIT: why there was an o already, who knows

**Comment by alex-eyre** [comment]
Sorry to be annoying, but this is no longer the most up-to-date version of awscli1, [1.19.72](https://github.com/aws/aws-cli/releases/tag/1.19.72) is, could that be the reason I'm getting this error trying to build it?:
```
ERROR: Could not find a version that satisfies the requirement colorama<0.4.4,>=0.2.5 (from awscli)
ERROR: No matching distribution found for colorama<0.4.4,>=0.2.5
```

**Comment by SuperSandro2000** [comment]
> colorama

We have version 0.4.4. Just substituteInPlace the version range.

**Comment by dotlambda** [comment]
Any updates on this?

**Comment by dotlambda** [comment]
Actually, these updates have all made it to master already.

**Pull request closed without merging**